ambry-protocol/src/main/java/com/github/ambry/protocol/AmbryRequests.java

```diff
@@ -683,7 +718,156 @@ public void handleReplicaMetadataRequest(NetworkRequest request) throws IOExcept
 
   @Override
   public void handleReplicateBlobRequest(NetworkRequest request) throws IOException, InterruptedException {
-    throw new UnsupportedOperationException("Request type not supported");
+    if (connectionPool == null || transformer == null) {
+      throw new UnsupportedOperationException("ReplicateBlobRequest is not supported on this node.");
+    }
+
+    ReplicateBlobRequest replicateBlobRequest;
+    if (request instanceof LocalChannelRequest) {
+      // This is a case where handleReplicateBlobRequest is called when frontends are talking to Azure. In this case, this method
+      // is called by request handler threads running within the frontend router itself. So, the request can be directly
+      // referenced as java objects without any need for deserialization.
+      replicateBlobRequest = (ReplicateBlobRequest) ((LocalChannelRequest) request).getRequestInfo().getRequest();
+    } else {
+      replicateBlobRequest = ReplicateBlobRequest.readFrom(new DataInputStream(request.getInputStream()), clusterMap);
+    }
+    long requestQueueTime = SystemTime.getInstance().milliseconds() - request.getStartTimeInMs();
+    long totalTimeSpent = requestQueueTime;
+    long startTime = SystemTime.getInstance().milliseconds();
+    metrics.replicateBlobRequestQueueTimeInMs.update(requestQueueTime);
+    metrics.replicateBlobRequestRate.mark();
+    ReplicateBlobResponse response = null;
+
+    // Get the two parameters from the replicateBlobRequest: the blobId and the remoteHostName
+    BlobId blobId = replicateBlobRequest.getBlobId();
+    String remoteHostName = replicateBlobRequest.getSourceHostName();
+    try {
+      Port remoteHostPort;
+      Optional<DataNodeId> remoteDataNode = blobId.getPartition().getReplicaIds().stream().map(ReplicaId::getDataNodeId)
+          .filter(node -> node.getHostname().equals(remoteHostName)).findFirst();
+      if (remoteDataNode.isPresent()) {
+        remoteHostPort = remoteDataNode.get().getPortToConnectTo();
+      } else {
+        logger.error("ReplicateBlobRequest {} couldn't find the remote host {} in the clustermap.", blobId, remoteHostName);
+        throw new IOException("ReplicateBlobRequest " + blobId + " couldn't find the remote host " + remoteHostName);
+      }
+      // ON_DEMAND_REPLICATION_TODO: Add configuration as replicationConfig.replicationConnectionPoolCheckoutTimeoutMs
+      final int connectionPoolCheckoutTimeoutMs = 1000;
+
+      List<PartitionRequestInfo> partitionRequestInfoList = new ArrayList<PartitionRequestInfo>();
+      List<BlobId> blobIds = new ArrayList<BlobId>();
+      blobIds.add(blobId);
+      PartitionRequestInfo partitionInfo = new PartitionRequestInfo(blobId.getPartition(), blobIds);
+      partitionRequestInfoList.add(partitionInfo);
+      // Get the Blob including expired and deleted.
+      final String clientId = ON_DEMAND_REPLICATION_CLIENTID_PREFIX + currentNode.getHostname();
```

Comment:
nit: May be a separator (-) between `replication-ondemand-fetch` prefix and hostname